kmm/kmm-post/src/commonTest/kotlin/com/gchristov/newsfeed/kmmpost/PostViewModelTest.kt

```diff
@@ -49,6 +50,7 @@ class PostViewModelTest : CommonViewModelTestClass() {
         ) { viewModel, postRepository ->
             // Then
             postRepository.assertCacheCleared()
+
```

Comment:
Lets analyse what this is doing and why it's causing issues.

View-model tests should be ensuring the correct state is set in various situations. This state is then rendered on the UI and verified separately using UI tests on the relevant clients (iOS, Android). 

The idea behind the cache test is to ensure that while we're loading/displaying the cache the UI still shows loading state and that the cache is actually shown when available. To achieve this, we:
- setup the test with some cached response
- setup the test with a `LoadsForever` response type for the API - if we don't have this, everything will happen synchonously when the test run finishes and we will never be able to verify that the screen is still loading while the cache is being shown. With this, we basically tell `FakeFeedRepository` to simulate a long-running API task, so that the view-model has time to show the cache while still keeping the loading state. Feel free to take a look at `FakeResponse.kt` for implementation details but basically there's a 10s artificial delay - we have no way to tell coroutines to load forever unless we block them manually. 
- run test with those settings
- after the test run is finished, we should expect the loading to still be visible (because the API is still loading)
- also, we should expect the cache to be set on the view-model state
- also, unrelated but important, we don't want to be showing any errors on the UI

Having said this, and without giving too many hints, if you take a look at your `DecorateUseCase` implementation, we can see that if there is a cached post we never actually call the API, which means that the `LoadsForever` will never be executed, so your unit test ends up in `loading == false` state.

Since the unit test and intended app functionality is to show cache first, then refresh the cache and then show the fetched post, this test fails correctly at this point. See my comment higher up for more explanation around the intended behaviour 🙂

Reply:
Done and clarified